### DISCUSSION TITLE
Open sourcing Laurel's project

### DISCUSSION THREAD

**Discussion by eddiejaoude** [EddieHubCommunity/support#3800] [Ideas]
⏰ LIVE stream 11th April Monday 2pm GMT
👥 Guests [Michael](https://github.com/dnsmichi) and [Laurel](laurelayounis)

## Goal

To open source the app https://talktometechnology.netlify.app

### Questions

- will there be more projects in the future (should we use a GitHub Org)?
- do you have maintainers you would like to add to the project?
- currently technologies
- future technologies
- GitHub Sponsors 

### Topics to cover in the live stream

- Technologies
- Issues + Labels
- Commits
- Documentation
- Automation with GitHub Actions (linting ... releases ... deployment)

**Comment by dnsmichi** [comment]
Thanks for starting the discussion @eddiejaoude :-)  

### Organisation

- Issue and PR templates
- .gitignore
- LICENSE
- Default labels, and review them
- Contributing.md
- README.md with the what, why and how
- Document the knowledge needed to get started with
   - Dev environments: Local, Docker, Gitpod
   - CI/CD - expand on automated tests, and deployments to Netlify 

### Technologies

It seems that it is Vanilla HTML/JS and there are plans to move to the MERN stack. 

I'd suggest the following for CI/CD

- Linting + security
- Prepare for MERN components and/or create issues directly in the new projects

General focus:

- CI for unit tests / lining / security
- CD for automated releases, packaging - maybe containers and registry usage
- CD for deploying to Netlify automatically. 

### Other thoughts

- Work with Pull Requests to suggest changes in the stream
- Prepare a set of issues to be created with task lists and action items

```
## Problem to solve

## Proposal

## Actions

- [ ] ...
- [ ] ...
- [ ] ... 
```

FYI, my working task is in https://gitlab.com/gitlab-com/marketing/corporate_marketing/corporate-marketing/-/issues/6199 and links to over here as single-source-of-truth. 

@laurelayounis I'd love for us to get access to the source code as tarball copy, to initially see the technology and potential blockers in the stream, we can do that on Thursday. Maybe you'll have a development documentation somewhere as well - any resource already here helps :-)

**Reply by eddiejaoude** [reply]
Awesome @dnsmichi 🔥

**Reply by laurelayounis** [reply]
Questions

**Will there be more projects in the future (should we use a GitHub Org)?**

Yes we would be interested in doing a GitHub Org. We would like to pursue continuing to build out additional open source accessibility projects and technologies to provide to the user base for free. Some future projects we're considering are: an emotional teaching and regulation application for people on the spectrum, and a digital aba skills trainer. 

**Link to the current Talk to Me Technology Repository:**
<https://github.com/laurelayounis/talktometechnology>


**do you have maintainers you would like to add to the project?**  
Matthew Mitchell <https://github.com/codebymatthew>


**Project Goals short term:**  
Make more responsive
Update images to be more reflective of words on the tiles
Various language translations
Enable custom tile creation
Add additional tiles of commonly used words such as "are"
Sentence builder
color themes

**Project Goals long term:**  
User accounts
customizable layouts
customizable voices
dialect functionality 


**Current Technologies.**  
Using HTML/CSS/Vanilla JS currently.  

**Future Technologies.**  
We will continue building out on the full MERN stack

**Reply by eddiejaoude** [reply]
Thanks Laurel 👍

**Comment by dnsmichi** [comment]
https://twitter.com/LaurelYounisDev/status/1511884081960685569

**Comment by eddiejaoude** [comment]
Repo https://github.com/EddieHubCommunity/support/discussions/3800

## Before live stream

1. [create GitHub Org](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/creating-a-new-organization-from-scratch)
2. move repo to GitHub Org
3. create team for repo (same name as repo)
4. add Eddie + Michael to the team as admin

## Live stream

1. hide repo features not being using (wiki, packages)
2. project board (kanban)
3. create issues of the tasks we need to do
   - license
   - code of conduct
   - readme + screenshot (include current and future technologies)
   - issue template
   - labels (start with default labels, then using open source standards)
   - contributing file
   - create release
4. quickstart
   - gitpod - example [LinkFree](https://github.com/EddieHubCommunity/LinkFree)
   - how to get the project running locally
5. hosting on GitHub Pages
6. GitHub Actions 
   - spell checker
   - lighthouse checker
   - releases - commits need to follow [Conventional Commit Message](https://www.conventionalcommits.org/en/v1.0.0/)
7. Future ideas
   - NextJS?

## Post live stream

1. remove Eddie + Michael from the GitHub Org team
2. documentation folder `docs/`

## Future

1. GitHub Action releases
2. Automated testing (Cypress + Cucumber), then Accessibility testing
3. Security testing
4. TwitterSpace about open sourcing a project

**Reply by eddiejaoude** [reply]
![Soccer Training Youtube Thumbnail Template-03](https://user-images.githubusercontent.com/624760/162237014-de2dad7e-2db0-470b-b3ba-66a5058b19e5.jpg)

**Reply by eddiejaoude** [reply]
<img width="1680" alt="Screenshot 2022-04-07 at 23 33 02" src="https://user-images.githubusercontent.com/624760/162250231-a588222a-4971-41b8-a0f0-bc41628f4714.png">

**Comment by laurelayounis** [comment]
Just updating the progress on what has been done: 
GitHub Org has been created: XollabOS
Talk to me technology repo has been moved to the XollabOS org
Eddie & Michael have been added to the org with full permissions (Please let me know if either of you don't have the full permissions & I'll get it fixed) 
It's been wonderful collaborating with everyone! Hope you have a great weekend. Looking forward to Monday's stream.

**Reply by eddiejaoude** [reply]
Thanks Laurel !! All looks good 👍 

I am excited for the live stream, see you all in 30mins 🎉

**Reply by dnsmichi** [reply]
@laurelayounis I think the maintainers team needs to be assigned to the repository in a maintainer role. 

<img width="1138" alt="image" src="https://user-images.githubusercontent.com/382049/162752091-2b1571bb-d391-4010-b137-729a8490b271.png">

With more permissions granted, one should see the `settings` tab which I cannot see atm. 

<img width="993" alt="image" src="https://user-images.githubusercontent.com/382049/162752233-e20f0298-5723-4282-8bda-83129aefc1b8.png">

 We can fix that in the stream though.

**Reply by eddiejaoude** [reply]
Good spot 👍 , yep we can sort out during the live stream!
